certification/internal/shell/scorecard_basic-check-spec.go

```diff
@@ -0,0 +1,174 @@
+    err = os.Chdir(tmpDir)
+    if err != nil {
+        logger.Error("unable to change dir: ", err)
+        return false, err
+    }
+    
+    configFile := filepath.Join(bundleDir,"/tests/scorecard/config.yaml")
+
+    //copy config file to the root
+    input, err := ioutil.ReadFile(configFile)
+    if err != nil {
+        logger.Error("unable to read config.yaml: ", err)
+        return false, err
+    }
+    
+    destinationFile := filepath.Join(tmpDir,"/config.yaml")
+    
+    err = ioutil.WriteFile(destinationFile, input, 0777)
+    if err != nil {
+        logger.Error("unable to write to config.yaml file: ", err)
+        return false, err
+    }
+    
+    _, err = exec.Command("chmod", "-R", "go+r", "./").CombinedOutput()
+    if err != nil {
+        logger.Error("unable to execute chmod: ", err)
+        return false, err
+    }
+    
+    logger.Debugf("Running Scorecard check for ",bundleImage)
+    
+    stdouterr, scorecarderr := exec.Command("operator-sdk", "scorecard",   
+                                  "--config", destinationFile,
+                                  "--selector=test=basic-check-spec-test",
+                                  "--output", "json", "bundle").CombinedOutput()
+    
+    scorecardFile := filepath.Join(artifactsDir,"/",scorecardBasicCheckResult)
+    
+    err = ioutil.WriteFile(scorecardFile, stdouterr, 0644)
+    if err != nil {
+        logger.Error("unable to copy result to /artifacts subdir: ", err)
+        return false, err
+    }
+
+    // we must send gojq a interface{}, so we have to convert our inspect output to that type
+    var inspectData interface{}
+    
+    err = json.Unmarshal(stdouterr, &inspectData)
+    if err != nil {
+        logger.Error("unable to parse scorecard json output")
+        logger.Debug("error Unmarshaling scorecard json output: ", err)
+        logger.Trace("failure in attempt to convert the raw bytes from `operator-sdk scorecard` to a interface{}")
+        return false, err
+    }
+
+    query, err := gojq.Parse(".items[].status.results[] | .name, .state")
+    if err != nil {
+        logger.Error("unable to parse scorecard json output")
+        logger.Debug("unable to parse :", err)
+        return false, err
+    }
+
+    // gojq expects us to iterate in the event that our query returned multiple matching values, but we only expect one.
+    iter := query.Run(inspectData)
+    
+    foundTestFailed := false
+
+    logger.Debug("scorecard output")
+    
+    for {
+        v, ok := iter.Next()
+        if !ok {
+            break
+        }
+        if err, ok := v.(error); ok {
+            logger.Error(err)
+        }
+        var s string = v.(string)
+        logger.Debug(s)
+        if strings.Contains(s, "fail") {
+            foundTestFailed = true
+        }
+    }
+
+    if foundTestFailed {
+        logger.Debug(" Found one or more tests failed the check.")
+        return false, nil
+    }
+
+    if scorecarderr != nil {
```

Comment:
I suggest you move this `if` block to line 93. Handling the erroneous condition must be done immediately.

Reply:
So scorecard can failed because one or more test failed to run. The output contains information pertaining to which test failed to run. For now, I store the output in artifacts/ and it's information the developer can use to fix the problems.
With that said, if I exit at line 93, I will have no way of digesting this information.